rusty_results/prelude.py

```diff
@@ -217,6 +217,29 @@ def unwrap_empty(self):
         """
         ...  # pragma: no cover
 
+    @abstractmethod
+    def flatten_one(self) -> Union[T, "Option[T]"]:
```

Comment:
I think this typing is accurate. By flattening we always go from Option to Option, we will never be returning a `T` itself, so that Union is a bit of a lie 😃 .

I think it should be something like this:
```python
def flatten_one(self: Option[Union[T, Option[T]]]) -> Option[T]:
...
```